tools/benchmark/memory/src/main.rs

```diff
@@ -112,11 +89,36 @@ fn get_meta_data(root_dir: &Path) -> Metadata {
 ///   d. Add the output to an `ndjson` file.
 ///   e. Move the dhat-heap.json file to the benchmark folder.
 fn main() {
-    let ProcessedArgs {
-        os,
-        examples,
-        toolchain,
-    } = process_cli_args();
+    let ProcessedArgs { os, examples } = process_cli_args();
+
+    let examples = if !examples.is_empty() {
+        examples
+    } else {
+        Command::new("cargo")
+            .arg("build")
+            .arg("--examples")
+            .arg("--features")
+            .arg("icu_benchmark_macros/benchmark_memory")
+            .arg("--features")
+            .arg("bench")
+            .stderr(Stdio::piped())
+            .spawn()
+            .unwrap_or_else(|err| {
+                eprintln!("Failed to collect examples {err:?}");
+                process::exit(1);
+            });
+        fs::read_dir("target/debug/examples")
+            .unwrap()
+            .flat_map(|entry| {
+                entry
+                    .ok()?
+                    .file_name()
+                    .into_string()
+                    .ok()
+                    .filter(|s| !s.contains(['-', '.']))
+            })
+            .collect()
+    };
```

Comment:
line 244

Reply:
> Also it seems this could be sensitive to old examples that were renamed but still have binaries in the target directory.

locally yes, but on CI we're doing clean builds